### PR TITLE
Add KXOR support to XYK pool

### PR DESCRIPTION
**Pull request by vovac12**

### CODE REVIEW

**Review by Alexey-N-Chernyshov — APPROVED** [review]

File: pallets/pool-xyk/src/utils.rs [hunk 1]
```diff
@@ -202,25 +192,47 @@ impl<T: Config> Pallet<T> {
...
+        };
+        let is_allowed_chameleon_pool = T::GetChameleonPool::get(&tpair);
+        if is_chameleon_pool && !is_allowed_chameleon_pool {
+            Err(Error::<T>::RestrictedChameleonPool)?
         }
+        Ok((tpair, base_chameleon_asset_id, is_allowed_chameleon_pool))
+    }
+
+    /// Sort assets into base and target assets of trading pair, if none of assets is base then return error.
+    pub fn strict_sort_pair(
```
Comment (wer1st) [inline pallets/pool-xyk/src/utils.rs:230]: Looks like function name & description are not relevant anymore

File: pallets/pool-xyk/src/utils.rs [hunk 2]
```diff
@@ -202,25 +192,47 @@ impl<T: Config> Pallet<T> {
...
-            })
-        } else if asset_b == base_asset_id {
-            Ok(TradingPair {
-                base_asset_id: asset_b.clone(),
-                target_asset_id: asset_a.clone(),
-            })
+        let base_chameleon_asset_id =
+            <T::GetChameleonPoolBaseAssetId as GetByKey<_, _>>::get(&base_asset_id);
+        let (ta, is_chameleon_pool) = if base_asset_id == asset_a {
+            (asset_b, false)
```
[the remark this answers is not in the document]
Comment (vovac12) [reply pallets/pool-xyk/src/utils.rs:206]: How?

[the remark this answers is not in the document]
Comment (wer1st) [reply pallets/pool-xyk/src/utils.rs:206]:
```rust
get_pair_info(XOR, XOR, GetChameleonPoolBaseAssetId::get())
```

[the remark this answers is not in the document]
Comment (vovac12) [reply pallets/pool-xyk/src/utils.rs:206]: It's expected behavior. XOR-KXOR swaps via XYK pool should be available and this pool is not a Chameleon pool

**Review by wer1st — APPROVED** [review]

**Comment by Nolan730** [comment]
What will happen when no xor left inside pull after there is constant sell of kxor and buy of xor? I feel like this will be a problem

**Pull request merged**